[JIT] add float[] str[] constants

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31503 [JIT] add float[] str[] constants**

Add support for float lists and string lists constants, which enables better constant propagation + constant pooling + freezing.

Differential Revision: [D19233558](https://our.internmc.facebook.com/intern/diff/D19233558)

Commit:
add float[] str[] constants

[ghstack-poisoned]

zdevito commented on test/test_jit.py:
```diff
@@ -3153,6 +3153,20 @@ def constant_prop(iter):
 
+    def test_constant_insertion(self):
+        def foo():
+            a = [1.0, 2.0, 3.0]
```
I'd double check that the python printer prints out string lists correctly.  I am pretty sure they printout without appropriate quoting.

zdevito reviewed (approved):
The code looks good, but I think printout out of string constant lists is probably broken in serialization but ends up untested _now_ because there is no way they can reach the printer. That might not always be true though, so we should make sure it is correct.

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 49f2965:


* **2/2** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=49f2965b484dbc0786de0472bac8ed43bf9e46ae).

### :detective: 2 new failures recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4026897) pytorch_linux_xenial_py3_6_gcc5_4_ge_config_simple_test (1/2)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=77975652) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=77975652))
<details>
<summary>
<code>Dec 20 01:14:46 RuntimeError: test_jit_simple failed!</code>
</summary>

```
Dec 20 01:14:46 Ran 2307 tests in 91.113s 
Dec 20 01:14:46  
Dec 20 01:14:46 FAILED (failures=1, errors=1, skipped=76, expected failures=1) 
Dec 20 01:14:46  
Dec 20 01:14:46 Generating XML reports... 
Dec 20 01:14:46 Traceback (most recent call last): 
Dec 20 01:14:46   File "test/run_test.py", line 456, in <module> 
Dec 20 01:14:46     main() 
Dec 20 01:14:46   File "test/run_test.py", line 449, in main 
Dec 20 01:14:46     raise RuntimeError(message) 
Dec 20 01:14:46 RuntimeError: test_jit_simple failed! 
Dec 20 01:14:46 =================== sccache compilation log =================== 
Dec 20 01:14:46 + cleanup 
Dec 20 01:14:46 + retcode=1 
Dec 20 01:14:46 + set +x 
Dec 20 01:14:46 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 20 01:14:46 Compile requests                56 
Dec 20 01:14:46 Compile requests executed        0 
Dec 20 01:14:46 Cache hits                       0 
Dec 20 01:14:46 Cache misses                     0 
Dec 20 01:14:46 Cache timeouts                   0 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4026899) pytorch_linux_xenial_py3_6_gcc5_4_ge_config_legacy_test (2/2)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=77975653) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=77975653))
<details>
<summary>
<code>Dec 20 01:15:18 RuntimeError: test_jit_legacy failed!</code>
</summary>

```
Dec 20 01:15:18 Ran 2307 tests in 136.167s 
Dec 20 01:15:18  
Dec 20 01:15:18 FAILED (failures=1, errors=1, skipped=58, expected failures=1) 
Dec 20 01:15:18  
Dec 20 01:15:18 Generating XML reports... 
Dec 20 01:15:18 Traceback (most recent call last): 
Dec 20 01:15:18   File "test/run_test.py", line 456, in <module> 
Dec 20 01:15:18     main() 
Dec 20 01:15:18   File "test/run_test.py", line 449, in main 
Dec 20 01:15:18     raise RuntimeError(message) 
Dec 20 01:15:18 RuntimeError: test_jit_legacy failed! 
Dec 20 01:15:18 =================== sccache compilation log =================== 
Dec 20 01:15:18 + cleanup 
Dec 20 01:15:18 + retcode=1 
Dec 20 01:15:18 + set +x 
Dec 20 01:15:18 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 20 01:15:18 Compile requests               304 
Dec 20 01:15:18 Compile requests executed        0 
Dec 20 01:15:18 Cache hits                       0 
Dec 20 01:15:18 Cache misses                     0 
Dec 20 01:15:18 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "[JIT] add float[] str[] constants"


Add support for float lists and string lists constants, which enables better constant propagation + constant pooling + freezing.

[ghstack-poisoned]

Commit:
Update on "[JIT] add float[] str[] constants"


Add support for float lists and string lists constants, which enables better constant propagation + constant pooling + freezing.

[ghstack-poisoned]

Commit:
Update on "[JIT] add float[] str[] constants"


Add support for float lists and string lists constants, which enables better constant propagation + constant pooling + freezing.

[ghstack-poisoned]

Commit:
Update on "[JIT] add float[] str[] constants"


Add support for float lists and string lists constants, which enables better constant propagation + constant pooling + freezing.

[ghstack-poisoned]

mrshenli commented:
@eellison  master starts to hit the following error since this PR, shall we revert?

https://app.circleci.com/jobs/github/pytorch/pytorch/4088679
https://app.circleci.com/jobs/github/pytorch/pytorch/4093221
https://app.circleci.com/jobs/github/pytorch/pytorch/4093698

```
Dec 31 03:10:12 test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime_opset10::test_interpolate_downsample Fatal Python error: Segmentation fault

Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/runner.py", line 178 in call_and_report
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/runner.py", line 95 in runtestprotocol
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/runner.py", line 80 in pytest_runtest_protocol
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/callers.py", line 187 in _multicall
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 86 in <lambda>
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 92 in _hookexec
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 258 in pytest_runtestloop
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/callers.py", line 187 in _multicall
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 86 in <lambda>
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 92 in _hookexec
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 237 in _main
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 193 in wrap_session
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 230 in pytest_cmdline_main
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/callers.py", line 187 in _multicall
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 86 in <lambda>
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 92 in _hookexec
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__
Dec 30 22:33:41   File "/usr/local/lib/python3.6/dist-packages/_pytest/config/__init__.py", line 90 in main
Dec 30 22:33:41   File "/usr/local/bin/pytest", line 11 in <module>
Dec 30 22:33:42 /var/lib/jenkins/workspace/scripts/onnx/test.sh: line 63: 17674 Segmentation fault      (core dumped) pytest "${args[@]}" "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py"
Exited with code exit status 139
```

eellison commented:
@mrshenli yep, D19203827 is revert diff
